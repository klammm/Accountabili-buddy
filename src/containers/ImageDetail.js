import React, { Component } from 'react';
import { Image, StyleSheet, Text, View } from 'react-native';
import { CardSection, Card } from '../components/common';
import Dimensions from 'Dimensions';

const ImageDetail = ({ image, uri }) => {
  const { imageUrl, caption } = image;
  const { imgContainer, imagesStyle, imageContainerStyle, postImage,
          postTestStyle, canvas, imageAvatar, cardStyle, cardSectionStyle } = styles;

  return (
    <Card style={cardStyle}>
    <CardSection style={cardSectionStyle}>
      <Image
        style = {imageAvatar}
        source= {{ uri: uri }}/>
    </CardSection>
    <CardSection style={cardSectionStyle}>
        <View style={styles.imgContainer}>
          <Image
            style={styles.imgCanvas}
            resizeMode='cover'
            resizeMode= {"contain"}
            source={{ uri: imageUrl }}
          />
        </View>
    </CardSection>
    <CardSection style={cardSectionStyle}>
        <Text style={postTestStyle}>{caption}</Text>
    </CardSection>
    </Card>
  );
};

const win = Dimensions.get('window');
const widthspec = win.width;
const hightspec = win.height - 30;
const white = 'rgba(255, 255, 255, 0.96)';
const purpleOpacity = 'rgba(127, 13, 205, 0.38)';
const greyOpacity = 'rgba(138, 138, 138, 0.38)';

const styles = StyleSheet.create({
  imgCanvas: {
      flex:1,
      width: null,
      height: null
  },
  imgContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    flex: 1
  },
  cardStyle: {
    backgroundColor: white,
    borderBottomWidth: 1,
    borderColor: greyOpacity,
    marginLeft: 0,
    marginRight: 0,
    marginTop: 0,
  },
  cardSectionStyle: {
    marginLeft: 10,
    padding: 5,
    marginTop: 10,
    marginBottom: 10
  },
  imageAvatar: {
    flexDirection: 'row',
    borderRadius: 14,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.4,
    shadowRadius: 2,
    elevation: 1,
    height: 30,
    width: 30,
    marginLeft: 10,
    backgroundColor: 'white'
  },
  postTestStyle :{
    marginLeft: 5
  }
});

export default ImageDetail;
