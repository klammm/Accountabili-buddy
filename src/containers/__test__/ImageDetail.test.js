import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ImageDetail } from '../ImageDetail';


describe('tests the ImageDetail component', () => {
  it('renders ImageDetail component', () => {
  const component = shallow(
    <ImageDetail
      image={{}}
    />
  );
    expect(toJson(component)).toMatchSnapshot();
  });
});
