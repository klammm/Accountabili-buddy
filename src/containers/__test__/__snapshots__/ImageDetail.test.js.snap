// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the ImageDetail component renders ImageDetail component 1`] = `
<Card>
  <CardSection>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "alignSelf": "stretch",
            "flex": 1,
            "height": 1304,
            "width": 720,
          }
        }
      />
    </View>
  </CardSection>
  <CardSection>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </CardSection>
</Card>
`;
